app/assets/javascripts/Components/Result/pdf_viewer.jsx

```diff
@@ -101,31 +124,64 @@ export class PDFViewer extends React.PureComponent {
     );
   };
 
-  rotate = () => {
-    annotation_manager.rotateClockwise90();
-    this.pdfViewer.rotatePages(90);
+  getZoomValuesToDisplayName = () => {
+    const zoomLevels = Array.from({length: 20}, (_, i) => ((i + 1) * 0.1).toFixed(1));
+
+    const valueToDisplayName = zoomLevels.reduce(
+      (acc, value) => {
+        acc[value] = `${(value * 100).toFixed(0)} %`;
+        return acc;
+      },
+      {"page-width": I18n.t("results.fit_to_page_width")}
+    );
+
+    return valueToDisplayName;
   };
 
   render() {
     const cursor = this.props.released_to_students ? "default" : "crosshair";
     const userSelect = this.props.released_to_students ? "default" : "none";
+    const zoomValuesToDisplayName = this.getZoomValuesToDisplayName();
+
     return (
-      <div className="pdfContainerParent">
-        <div
-          id="pdfContainer"
-          className="pdfContainer"
-          style={{cursor, userSelect}}
-          ref={this.pdfContainer}
-        >
-          <div id="viewer" className="pdfViewer" />
+      <React.Fragment>
+        <div className="toolbar">
+          <div className="toolbar-actions">
+            {I18n.t("results.current_rotation", {rotation: this.state.rotationInDegrees})}
+            <button onClick={this.rotate} className={"inline-button"}>
+              {I18n.t("results.rotate_image")}
+            </button>
+            <span style={{marginLeft: "7px"}}>{I18n.t("results.zoom")}</span>
+            <SingleSelectDropDown
+              valueToDisplayName={zoomValuesToDisplayName}
+              options={Object.keys(zoomValuesToDisplayName)}
+              selected={this.state.zoom}
+              dropdownStyle={{minWidth: "auto", marginLeft: "5px", width: "150px"}}
```

Comment:
The width is still a bit awkward here; try `fit-content` instead. Ideally this would be applied to all of the options in the expanded dropdown (right now the options seem to have a fixed width that's larger than necessary)